gtsfm/runner/gtsfm_runner_base.py

```diff
@@ -86,11 +106,30 @@ def construct_scene_optimizer(self) -> SceneOptimizer:
 
         return scene_optimizer
 
+    def construct_retriever(self):
+        """Set up retriever module."""
+        if self.parsed_args.matching_regime == "exhaustive":
+            retriever = SequentialRetriever(max_frame_lookahead=MAX_POSSIBLE_FRAME_LOOKAHEAD)
+
+        elif self.parsed_args.matching_regime == "retrieval":
+            retriever = NetVLADRetriever(num_matched=self.parsed_args.num_matched)
+
+        elif self.parsed_args.matching_regime == "sequential":
+            retriever = SequentialRetriever(max_frame_lookahead=self.parsed_args.max_frame_lookahead)
+
+        elif self.parsed_args.matching_regime == "sequential_with_retrieval":
+            retriever = JointNetVLADSequentialRetriever(
+                num_matched=self.parsed_args.num_matched, max_frame_lookahead=self.parsed_args.max_frame_lookahead
+            )
+        return retriever
```

Comment:
Maybe the config would be a better place for this?

Reply:
This was something I went back and forth on. If we want it configured by the config, it would need to go into the SceneOptimizer class. But we run most of the SceneOptimizer with dask, whereas this would not be run with Dask.

The other thing was that I wanted this to be mainly configured by CLI for flexibility.

But I'm open to rethinking this.